bolt-modules/ctrl/lib/puppet/functions/ctrl/do_until.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# Repeat the block until it returns 'true'.
+Puppet::Functions.create_function(:'ctrl::do_until') do
+  # @example Run a task until it succeeds
+  #   ctrl::do_until || {
+  #     run_task('test', $target, _catch_errors => true).ok?
+  #   }
+  dispatch :do_until do
+    block_param
+    return_type 'Undef'
+  end
+
+  def do_until
+    until yield; end
+    nil
```

Comment:
It might be useful to return the value on this one since it could be anything truthy. It might also behave unexpectedly....

Reply:
Ah, that could be useful.

Reply:
I agree but am not sure. I'll ask for feedback in slack

Reply:
Explore how `break` works in these, and whether it can be used to return a value. If we can use break to return a value, a loop might be a better alternative.

Reply:
Explore loop-recur:
```
loop || {
  <do something>
  if true { break }
}
```
or
```
loop(0) |$updated_val| {
  if $updated_val > 2 { break $updated_val }
  $updated_val + 1
}
```

Reply:
Break can't return a value. So I lean towards continuing with `do_while` and `do_until`.

Reply:
`break` works, but has the interesting side-effect of causing `do_until` to return `"break"`.